src/skip.rs

```diff
@@ -7,6 +7,7 @@ use rustc_ast_pretty::pprust;
 /// by other context. Query this context to know if you need skip a block.
 #[derive(Default, Clone)]
 pub(crate) struct SkipContext {
+    pub(crate) all_macros: bool,
     macros: Vec<String>,
```

Comment:
I don't really like this structure, I think it should be refactored into something like

```rust
enum SkipMacroContext {
    All,
    Named(HashSet<MacroName>),
}

pub(crate) struct SkipContext {
    macros: SkipMacroContext,
    attributes: HashSet<String>,
}
```

but thought that was probably out of scope for this PR. Happy to open a followup afterwards, or include as an additional commit if you think it's okay to add here.

Reply:
I think the PR is in a pretty good state and I'm leaning towards making this change in a follow up PR. If you feel strongly about it though and don't think it will add a lot of extra work to the current PR, then I'll leave that call up to you